spa/src/components/content/pages/Pages.js

```diff
@@ -81,17 +83,26 @@ function Pages() {
 
   const pagesByRevenueProgram = pagesbyRP(pages, pageSearchQuery);
 
+  const addPageButtonShouldBeDisabled = () => {
+    if ([USER_ROLE_HUB_ADMIN_TYPE, USER_SUPERUSER_TYPE].includes(user?.role_type?.[0])) {
+      return false;
+    }
+    const pageLimit = user?.organizations?.[0]?.plan?.page_limit ?? 0;
+    return pages.length + 1 > pageLimit;
+  };
+
   return (
     <>
       <GenericErrorBoundary>
+        {isLoading && <GlobalLoading />}
         <Hero
           title="Pages"
           subtitle="Welcome to Pages. Here you can create, manage, and publish contribution pages. Create a new page by selecting the ‘New Page’ button below."
           placeholder="Pages"
           onChange={setPageSearchQuery}
         />
         <Content data-testid="pages-list">
-          <NewButton onClick={handleOpen} />
+          <NewButton data-testid="new-page-button" disabled={addPageButtonShouldBeDisabled()} onClick={handleOpen} />
```

Comment:
Than here you don't have to call the function
```suggestion
          <NewButton data-testid="new-page-button" disabled={addPageButtonShouldBeDisabled} onClick={handleOpen} />
```

Reply:
Another comment, why do we need `data-testid` in all these places? As the components are accessible, we can find them all in cypress/jest using a11y getters

Reply:
That's the established pattern we have with Cypress. I agree we could take an a11y-centric approach for our Cypress tests, but for now I'm inclined to stick with existing approach for sake of consistency, keeping in mind that we've recently added a bunch of testing-library unit tests, which are inherently a11y-centric.